src/ValueObjects/Messages/Support/Media.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace Prism\Prism\ValueObjects\Messages\Support;
+
+use finfo;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Storage;
+use InvalidArgumentException;
+use Prism\Prism\Concerns\HasProviderMeta;
+
+class Media
+{
+    use HasProviderMeta;
+
+    protected ?string $localPath = null;
+
+    protected ?string $storagePath = null;
+
+    protected ?string $url = null;
+
+    protected ?string $rawContent = null;
+
+    protected ?string $base64 = null;
+
+    protected ?string $mimeType = null;
+
+    /**
+     * @deprecated Use `fromLocalPath` instead.
+     */
+    public static function fromPath(string $path): static
+    {
+        return self::fromLocalPath($path);
+    }
+
+    public static function fromLocalPath(string $path): static
+    {
+        if (! is_file($path)) {
+            throw new InvalidArgumentException("$path is not a file");
+        }
+
+        $content = file_get_contents($path);
+
+        if ($content === '' || $content === '0' || $content === false) {
```

Comment:
content could be one of these: - `''`, `'0'`, `0`, `0.0`, `null`, `false`, `[]`
```suggestion
        if (empty($content)) {
```

Reply:
https://laravel.com/docs/11.x/filesystem#file-metadata -> `$mime = Storage::mimeType('file.jpg');`

Reply:
The extended if is probably from the rector that transforms empty() into those three assertions.

laravel's blank() helper will do the job

Reply:
```
blank('');           // true
blank('0');          // false -> here if content has 0 it will be considered not empty
blank(null);         // true
blank([]);           // true
blank(collect([]));  // true
blank(0);            // false -> here too
```
If this is ok than I guess to use `blank()` make sense here

Reply:
Yeah sorry, should have noted the different behavior - you can think of blank as a "strict" empty

Reply:
So many versions of empty! Prism doesn't actually have that rector, but I assumed it did from looking at the existing code. So we can just use `empty(trim($content))`

Reply:
Hahah well, that's awkward, I assumed on that basis, too. I like your suggestion, the `blank` helper always looks a little crowded to me. Less useless checks & no dependency 🎉

Reply:
I take this back. Rector wants one of these (depending on situation) instead of my suggestion 😆 

```php
in_array(trim($content), ['', '0'], true)
```

```php
in_array(trim($content ?? ''), ['', '0'], true)
```

I'm going to comply with it!